src/bm_sim/calculations.cpp

```diff
@@ -310,6 +311,12 @@ struct crcCCITT {
 
 struct cksum16 {
   uint16_t operator()(const char *buf, size_t len) const {
+#ifdef BM_LOG_DEBUG_ON
+    std::cout << "Printing data: " << len << std::endl;
+    for (size_t i = 0; i < len; i++)
+      fprintf(stderr, "%02X", buf[i]);
+    std::cout << "\nEnd\n";
+#endif
```

Comment:
remove?

this is not helpful to users, not to mention you are just outputting to `std::cout`

Reply:
Would be good to not remove.

An IPv6 layer-4 checksum P4 example in p4c is still pending to test with simple_switch.  Further, there are 4-5 different ways to write P4 code for layer-4 options.  If a discrepancy is found in checksum,  a user would like to see the data simple_switch used to computer the checksum.  I will change the `std::cout` to qualified debug.

Is there a preferred way to enable BM_LOG_DEBUG_ON in behavioral-mode build?  First thought I had was to add 
`-DBM_LOG_DEBUG_ON` to Makefile `DEFS`